### ISSUE TITLE
no-useless-path-segments rule only works with paths which name starts with "."

### ISSUE THREAD

**Issue by aliaksandr-yermalayeu** [benmosher/eslint-plugin-import#1280]
Given the following folder structure:

```
project-root
├── .eslintrc.js
├── package.json
└── src
    ├── index.js
    └── module.js
```

package.json contents:
```json
{
  "devDependencies": {
    "eslint": "^5.13.0",
    "eslint-plugin-import": "^2.16.0"
  },
  "dependencies": {
    "lodash": "^4.17.11"
  }
}
```

.eslintrc contents:
```json
module.exports = {
    "plugins": [
        "import"
    ],
    "rules": {
        "import/no-useless-path-segments": "error"
    },
    "env": {
        "browser": true,
        "es6": true
    },
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "root": true
};
```

index.js contents:
```javascript
import sumBy from "lodash/sumBy"; //this is ok
import foo from ".//a.js"; // this is causing error
```

however the following examples don't cause the error:
```javascript
import sumBy from "lodash/////sumBy"; //too many slashes

import sumBy from "lodash/fn/../sumBy"; //should be "lodash/sumBy"

import sumBy from "lodash//fn/..//sumBy"; //should be "lodash/sumBy"

import foo from "src//a.js";  // absolute path, should be "src/a.js"

import foo from "src/bar/../a.js";  // absolute path, should be "src/a.js"
```

It might be due to the next code:
https://github.com/benmosher/eslint-plugin-import/blob/bdc05aa1d029b70125ae415e5ca5dca22250858b/src/rules/no-useless-path-segments.js#L70-L72

My expectation is that the rule should work for all imports, not only those starting with dot.

**Comment by ljharb** [comment]
That makes sense - however, i suspect it will cause issues with `npm link` if we just remove that bailout.

**Comment by sergei-startsev** [comment]
@ljharb Can we hide it behind an option?

**Comment by ljharb** [comment]
That’s a good mitigation but it’d still need to work properly when enabled.

With or without an option, it’d be good to evaluate a PR that adds many test cases, including some with symlinks.

**Comment by SimonSomlai** [comment]
What also worked for me to disallow index imports is to use;

```tsx
    'no-restricted-imports': [
      'error',
      {
        patterns: ['./*', '../*', '*/**/index'],
      },
    ],
```

**Comment by YoussefYN** [comment]
Hello,

Are there any updates regarding this issue?
